src/module.c

```diff
@@ -145,10 +145,36 @@ int AddDocumentCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
   LG_DEBUG("Adding doc %s with %d fields\n", RedisModule_StringPtrLen(doc.docKey, NULL),
            doc.numFields);
   RSAddDocumentCtx *aCtx = NewAddDocumentCtx(sp, &doc);
+
+  // in partial mode
+  if (partial && replace) {
+    RedisSearchCtx sctx = {.redisCtx = ctx, .spec = sp};
+
+    // No indexable fields are updated, we can just update the metadata.
+    // Quick update just updates the score, payload and sortable fields of the document.
+    // Thus full-reindexing of the document is not required
+    if (!(aCtx->stateFlags & ACTX_F_INDEXABLES)) {
+      const char *errStr = NULL;
+      if (Document_QuickUpdate(&sctx, &doc, &errStr) == REDISMODULE_ERR) {
+        RedisModule_ReplyWithError(ctx, errStr ? errStr : "Error updating document");
+        goto errclean;
+      }
+      return RedisModule_ReplyWithSimpleString(ctx, "OK");
+    } else {
+
+      // The document update is partial - but it contains indexable fields. We need to load the
+      // current version of it, and then do a normal indexing flow
+      Redis_LoadDocument(&sctx, doc.docKey, &doc);
```

Comment:
This line has no effect. `&doc` here is never used by aCtx. Should I add an appropriate function in aCtx to handle the state? this is required because it makes assumptions and such based on the state of the document it was provided with.

This would all be easier if the logic was just moved to `Submit()` of course :)

Reply:
sure, just fix it.